value/src/main/java/com/google/auto/value/processor/AutoValueProcessor.java

```diff
@@ -397,6 +387,26 @@ private void processType(TypeElement type) {
       }
     }
 
+    if (appliedExtensions.size() > 0) {
+      final Set<String> methodsToRemove = Sets.newHashSet();
+      for (int i = appliedExtensions.size() - 1; i >= 0; i--) {
+        AutoValueExtension extension = appliedExtensions.get(i);
+        Collection<String> consumed = extension.consumeProperties(context);
+        if (consumed != null) {
```

Comment:
`null` collections make me :cry: You're also using it in the test. Perhaps make `AutoValueExtension` an abstract class with a default implementation that returns `Collections.emptySet()` and forbid `null`?